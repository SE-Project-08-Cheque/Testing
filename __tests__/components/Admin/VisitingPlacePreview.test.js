// import dependencies
import React from "react";

// import react-testing methods
import { render, fireEvent, cleanup } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";

// import react dom
import ReactDOM from "react-dom";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import VisitingPlacePreview from "../../../components/Admin/VisitingPlacePreview";

afterEach(cleanup);


it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <VisitingPlacePreview data={{ placename: "Galle", methods: ["bus"] }} />,
    div
  );
});

it("renders visiting place preview correctly", () => {
  const { getByTestId } = render(
    <VisitingPlacePreview
      data={{
        placeName: "Galle",
        distance: "20",
        timeToReach: "30",
        description: "no description",
        methods: ["bus"],
      }}
    />
  );
  expect(getByTestId("placename")).toHaveTextContent("Galle");
  expect(getByTestId("distance")).toHaveTextContent("20 km");
  expect(getByTestId("timetoreach")).toHaveTextContent("30 min");
  expect(getByTestId("description")).toHaveTextContent("no description");
});

it("edit button works correctly", () => {
  const { queryByTestId, getByText } = render(
    <VisitingPlacePreview
      data={{
        placeName: "Galle",
        distance: "20",
        timeToReach: "30",
        description: "no description",
        methods: ["bus"],
      }}
    />
  );

 
  expect(queryByTestId("placename")).toBeInTheDocument();
  expect(queryByTestId("distance")).toBeInTheDocument();
  expect(queryByTestId("timetoreach")).toBeInTheDocument();
  expect(queryByTestId("description")).toBeInTheDocument();

  expect(queryByTestId("placename_input")).not.toBeInTheDocument();
  expect(queryByTestId("distance_input")).not.toBeInTheDocument();
  expect(queryByTestId("timetoreach_input")).not.toBeInTheDocument();
  expect(queryByTestId("description_input")).not.toBeInTheDocument();
  
  fireEvent.click(getByText("Delete the place")) 
  fireEvent.click(getByText("Edit the place")) 

  expect(queryByTestId("placename")).not.toBeInTheDocument();
  expect(queryByTestId("distance")).not.toBeInTheDocument();
  expect(queryByTestId("timetoreach")).not.toBeInTheDocument();
  expect(queryByTestId("description")).not.toBeInTheDocument();

  expect(queryByTestId("placename_input")).toBeInTheDocument();
  expect(queryByTestId("distance_input")).toBeInTheDocument();
  expect(queryByTestId("timetoreach_input")).toBeInTheDocument();
  expect(queryByTestId("description_input")).toBeInTheDocument();
  
  fireEvent.click(getByText("Bus")) 
  fireEvent.click(getByText("Car")) 
  fireEvent.click(getByText("Foot")) 


  fireEvent.change(queryByTestId("placename_input"),{ target: { value: 'mirissa' } })
  fireEvent.change(queryByTestId("latitude_input"),{ target: { value: '23' } })
  fireEvent.change(queryByTestId("longitude_input"),{ target: { value: '23' } })
  fireEvent.change(queryByTestId("description_input"),{ target: { value: 'agasghasg hasdh' } })

 
  fireEvent.click(getByText("Update the place")) 

  
});

it("matches snapshot", () => {
  const tree = renderer.create(
    <VisitingPlacePreview
      data={{
        placename: "Galle",
        distance: "20",
        timeToReach: "30",
        description: "no description",
        methods: ["bus"],
      }}
    />
  );

  expect(tree).toMatchSnapshot();
});
