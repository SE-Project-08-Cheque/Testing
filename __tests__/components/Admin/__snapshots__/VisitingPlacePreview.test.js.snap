// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-mccp0p"
>
  <form
    action="submit"
  >
    <h4
      className="chakra-heading css-biiv2x"
      data-testid="placename"
    />
    <div
      className="css-1poympq"
    >
      <img
        alt="photo"
        className="chakra-image__placeholder css-clx8i5"
      />
    </div>
    <div
      className="chakra-stack css-1n2jfo2"
    >
      <div
        className="css-nc9qvg"
      >
        <p
          className="chakra-text css-1tapsqs"
        >
          Description
        </p>
        <p
          className="chakra-text css-0"
          data-testid="description"
        >
          no description
        </p>
      </div>
      <div
        className="css-nc9qvg"
      >
        <div
          className="css-0"
          data-testid="map"
        >
          <div
            style={
              Object {
                "cursor": "grab",
                "height": 400,
                "position": "relative",
                "width": "100%",
              }
            }
          >
            <div
              style={
                Object {
                  "height": "100%",
                  "position": "relative",
                  "width": "100%",
                }
              }
            >
              <div
                className=""
                style={
                  Object {
                    "height": "100%",
                    "overflow": "hidden",
                    "position": "absolute",
                    "visibility": "inherit",
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-nc9qvg"
      >
        <p
          className="chakra-text css-1n49pl7"
        >
          Distance
          <span
            className="chakra-badge css-1yz77up"
            data-testid="distance"
          >
            20
             km
          </span>
        </p>
      </div>
      <div
        className="css-nc9qvg"
      >
        <p
          className="chakra-text css-1n49pl7"
        >
          Travelling Method
          <span
            className="chakra-badge css-1e8ypgz"
          >
            bus
          </span>
        </p>
      </div>
      <div
        className="css-nc9qvg"
      >
        <p
          className="chakra-text css-1n49pl7"
        >
          Time to Reach
          <span
            className="chakra-badge css-1yz77up"
            data-testid="timetoreach"
          >
            30
             min
          </span>
        </p>
      </div>
      <div
        className="css-gmuwbf"
      >
        <div
          className="chakra-stack css-149v5c4"
        >
          <button
            className="chakra-button css-oqfhsh"
            onClick={[Function]}
            type="button"
          >
            Edit the place
          </button>
          <button
            className="chakra-button css-oqfhsh"
            onClick={[Function]}
            type="button"
          >
            Delete the place
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
